From 0724a00b779d411375886bdefcc4153adbcd3780 Mon Sep 17 00:00:00 2001
From: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Date: Wed, 18 May 2011 18:42:20 -0400
Subject: [PATCH 138/248] marker-fix-rcu

markers call_rcu usage: initialize rcu_head structures

Initialize rcu_head structures before passing them to call_rcu() to eliminate
false positives in DEBUG_RCU_HEAD.

Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@polymtl.ca>
CC: "Paul E. McKenney" <paulmck@linux.vnet.ibm.com>
CC: mingo@elte.hu
CC: akpm@linux-foundation.org
---
 kernel/marker.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/kernel/marker.c b/kernel/marker.c
index 0b009a9..9885cea 100644
--- a/kernel/marker.c
+++ b/kernel/marker.c
@@ -451,6 +451,7 @@ static struct marker_entry *add_marker(const char *channel, const char *name,
 	e->format_allocated = 0;
 	e->refcount = 0;
 	e->rcu_pending = 0;
+	INIT_RCU_HEAD(&e->rcu);
 	hlist_add_head(&e->hlist, head);
 	return e;
 }
-- 
1.7.0.4

